### ISSUE TITLE
how to modify anchor box?

### ISSUE THREAD

**Issue by cjt222** [fizyr/keras-retinanet#421]
hello，i want to modify ratio of anchor(i need 7 ratio),but when i modify anchor in util/anchor.py,it cause some problem,what parameters should i modily?
![image](https://user-images.githubusercontent.com/17508662/39111009-ee68903c-4705-11e8-9dae-c32d4dbc933f.png)

**Comment by hgaiser** [comment]
Currently there is no easy method to configure this (see https://github.com/fizyr/keras-retinanet/issues/324). Anchors are generated in two places unfortunately, once for creating the target data (which is the one you edited) and once for inference, which is the one you didn't edit: https://github.com/fizyr/keras-retinanet/blob/master/keras_retinanet/layers/_misc.py#L31-L45

**Comment by cjt222** [comment]
thank you for your answer， i have  also changed file in keras_retinanet/vim keras_retinanet/models/retinanet.py，it is ok!!!
![image](https://user-images.githubusercontent.com/17508662/39125913-363914ea-4733-11e8-9d05-0ac926e937b9.png)

**Comment by hgaiser** [comment]
Ah that would work for training, but it would still break when doing inference. It reminded me that my previous comment was wrong though, for inference it is better to create an AnchorParameters object like this : 
```python
anchor_parameters = keras_retinanet.models.retinanet.AnchorParameters(
    sizes   = [32, 64, 128, 256, 512],
    strides = [8, 16, 32, 64, 128],
    ratios  = np.array([0.5, 1, 2], keras.backend.floatx()),  # Change this for your ratios
    scales  = np.array([2 ** 0, 2 ** (1.0 / 3.0), 2 ** (2.0 / 3.0)], keras.backend.floatx()),  # Change this for your scales
)
```

Which you then should be able to pass to [this line](https://github.com/fizyr/keras-retinanet/blob/master/keras_retinanet/bin/train.py#L86) like so:

```python
 model = model_with_weights(backbone_retinanet(num_classes, backbone=backbone, modifier=modifier, anchor_parameters=anchor_parameters), weights=weights, skip_mismatch=True)
```

I haven't tried this yet though, so I don't know if it works or not..

**Comment by cjt222** [comment]
ok， i see, thank you so much!!!!

**Comment by whyguu** [comment]
I am wondering whether I can just modify AnchorParameter.default in retinanet.py to make all things work well. and I have tried and no errors occur, so whether there are some potential risks?@hgaiser

**Comment by whyguu** [comment]
I only change anchor ratios numbers from 3 to 4, and change retinanet(num_anchors=AnchorParameter.default.num_anchors()). i guess when I use imagenet_weights to train the model, there should be an error, because I do not change anchor.py, the labels generated can not match model out. BUT, there is no error happening. can someone get me rid of this trouble.
